server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -1435,6 +1435,8 @@ public SearchPhase newSearchPhase(
             SearchResponse.Clusters clusters
         ) {
             if (preFilter) {
+                // only for aggs we need to contact shards even if there are no matches
+                boolean requireAtLeastOneMatch = searchRequest.source().aggregations() != null;
```

Comment:
I believe our code doesn't  _really_ ensure the that `searchRequest.source()` is always non-null. We check for null across the codebase. Shall we add a null check here before calling `aggregations()` ?

Reply:
Nice catch :D yes, I'll add that!
About CI I'll wait for multiple green runs before merging it :D